serde-support/src/main/java/io/micronaut/serde/support/DefaultSerdeRegistry.java

```diff
@@ -411,10 +411,14 @@ public <T> Serde<T> findInternalSerde(Argument<T> type) {
 
     private void registerBuiltInSerdes() {
         Serdes.register(serdeConfiguration, introspections, serdeRegistrar -> {
-            for (Argument<?> type : serdeRegistrar.getTypes()) {
-                deserializers.add(new InternalSerdeBeanDefinition<>(type, Deserializer.class, serdeRegistrar, serdeRegistrar.getOrder()));
-                serializers.add(new InternalSerdeBeanDefinition<>(type, Serializer.class, serdeRegistrar, serdeRegistrar.getOrder()));
-                internalSerdes.add(new InternalSerdeBeanDefinition<>(type, Serde.class, serdeRegistrar, serdeRegistrar.getOrder()));
+            try {
+                for (Argument<?> type : serdeRegistrar.getTypes()) {
+                    deserializers.add(new InternalSerdeBeanDefinition<>(type, Deserializer.class, serdeRegistrar, serdeRegistrar.getOrder()));
+                    serializers.add(new InternalSerdeBeanDefinition<>(type, Serializer.class, serdeRegistrar, serdeRegistrar.getOrder()));
+                    internalSerdes.add(new InternalSerdeBeanDefinition<>(type, Serde.class, serdeRegistrar, serdeRegistrar.getOrder()));
+                }
+            } catch (NoClassDefFoundError ignore) {
```

Comment:
won't this break all serdes? We need to only skip SQL types

Reply:
This is invoked per serde so it will fail on sql types only. You can see the referenced issue that the exception is coming from `getTypes`